SubjectsDAL/Repos/BaseRepo.cs

```diff
@@ -43,8 +43,32 @@ internal int SaveChanges()
             }
         }
 
-        public T GetOne(int id) => table.Find(id);
+        public T GetOne(int? id) => table.Find(id);
```

Comment:
Why are you want to receive here nullable integer? This is the required parameter as I read the whole code and you should consider securing that in another way, before calling this method.